### PR TITLE
Correctly display lists and nullables in the runtime type domain

### PR DESCRIPTION
**Pull request by schrockn**
Previous type system refactor had introduced a regression when input type and output types of solid were listed as "Unnamed Type" . This adds a server side property to render [] and ? for lists and nullables. This is a bit of a stop gap. I think we will have to switch this to be client-side to get all the rendering goodness we want, but this is an improvement for now.

### CODE REVIEW

**Comment by schrockn** [comment]
before
screenshot 2019-01-30 08 06 47

**Comment by schrockn** [comment]
after:

screenshot 2019-01-30 08 08 04

File: python_modules/dagster/dagster/core/types/runtime.py [hunk 1]
```diff
@@ -227,6 +231,10 @@ def __init__(self, inner_type):
         )
         self.inner_type = inner_type
 
+    @property
+    def display_name(self):
+        return self.inner_type.display_name + '?'
```
Comment (bengotow) [inline python_modules/dagster/dagster/core/types/runtime.py]: Ahh this is cool - I think that we may want to leverage this new display_name idea in the configType schema viewer as well. Right now it gets `isList` and uses that to add the `[`, `]`, etc. client side.

File: js_modules/dagit/src/TypeWithTooltip.tsx [hunk 2]
```diff
@@ -17,19 +18,23 @@ export default class TypeWithTooltip extends React.Component<
...
         description
       }
     `
   };
 
   render() {
-    const { name } = this.props.type;
-    const search = `?typeExplorer=${name}`;
+    const { name, displayName } = this.props.type;
+    const search = `?typeExplorer=${displayName}`;
```
Comment (bengotow) [inline js_modules/dagit/src/TypeWithTooltip.tsx]: I think this might not work if the type is optional and it appends a `?`?

**Pull request merged**